tests/transforms/test_set_shots.py

```diff
@@ -195,3 +198,68 @@ def circuit(x):
 
         # Parameter-shift rule requires 2*num_params + 1 = 3 executions
         assert len(tracker_shots.history["executions"]) == 3
+
+    @pytest.mark.system
+    def test_userwarnings(self):
+        """
+        Test that using set_shots with a QNode that has a shots value
+        raises a UserWarning.
+        """
+
+        @qml.qnode(qml.device("default.qubit"))
+        def c():
+            return qml.sample(wires=0)
+
+        with pytest.warns(
+            UserWarning,
+            match=r"The transform will take precedence",
+        ):
+            set_shots(c, shots=10)(shots=20)
+
+    @pytest.mark.system
```

Comment:
Not entirely sure this test is necessary, as long as we have the above test.

Reply:
Yeah from the codebase functionality the second one is totlaly removable. It's only validating the more detailed user call sentence is poped out.